src/lexer.cc

```diff
@@ -0,0 +1,139 @@
+/*
+   Copyright 2019 Simon (psyomn) Symeonidis
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+#include "bk/lexer.h"
+#include "bk/token.h"
+
+#include <filesystem>
+#include <sstream>
+#include <fstream>
+
+namespace bk {
+static bool isPunct(char c) {
+  return
+    c == '{' || c == '}' ||
+    c == '.' || c == ',' ||
+    c == ';' || c == ':' ||
+    c == '\'' || c == '"' ||
+    c == '!' || c == '?' ||
+    c == '-' ||
+    c == '(' || c == ')' ||
+    c == '[' || c == ']' ||
+    c == '<' || c == '>';
```

Comment:
This is a matter of personal preference but you could also write this as:
```
static std::string pattern = "{}.,;:";
return pattern.find_first_of(c) != std::npos;
```

Reply:
Actually I think there's a cleaner version of `std::any_of` on an `std:array` or a plain old C string but I'd need to experiment a bit to figure out the cleanest way to write it. The upshot of the above solutions would be avoiding allocating a string and dealing with complex static constructors.